diff --git a/highlevel/methods/files/normalize-input-media.js b/highlevel/methods/files/normalize-input-media.js
index a735241111fa517ecd4ee03ad8ee67221ce24720..d42e97ccaa3f9c18ce0122f431b610a5867cd7df 100644
--- a/highlevel/methods/files/normalize-input-media.js
+++ b/highlevel/methods/files/normalize-input-media.js
@@ -276,7 +276,7 @@ async function _normalizeInputMedia(client, media, params = {}, uploadMedia = fa
           fileReference: res.photo.fileReference
         },
         ttlSeconds: media.ttlSeconds,
-        spoiler: media.type === "video" && media.spoiler,
+        spoiler: media.spoiler,
         livePhoto: Boolean(livePhotoVideo),
         video: livePhotoVideo
       };
