diff --git a/node_modules/@mtcute/core/highlevel/methods/files/normalize-input-media.js b/node_modules/@mtcute/core/highlevel/methods/files/normalize-input-media.js
index ecc6e83..bdd0ffd 100644
--- a/node_modules/@mtcute/core/highlevel/methods/files/normalize-input-media.js
+++ b/node_modules/@mtcute/core/highlevel/methods/files/normalize-input-media.js
@@ -250,7 +250,7 @@ async function _normalizeInputMedia(client, media, params = {}, uploadMedia = fa
           fileReference: res.photo.fileReference
         },
         ttlSeconds: media.ttlSeconds,
-        spoiler: media.type === "video" && media.spoiler
+        spoiler: (media.type === "video" || media.type === "photo") && media.spoiler
       };
     }
     assertTypeIs("normalizeInputMedia (@ messages.uploadMedia)", res, "messageMediaDocument");
@@ -264,7 +264,7 @@ async function _normalizeInputMedia(client, media, params = {}, uploadMedia = fa
         fileReference: res.document.fileReference
       },
       ttlSeconds: media.ttlSeconds,
-      spoiler: media.type === "video" && media.spoiler,
+      spoiler: (media.type === "video" || media.type === "photo") && media.spoiler,
       videoCover,
       videoTimestamp: media.type === "video" ? media.timestamp : void 0
     };
diff --git a/node_modules/@mtcute/core/tl/inner-tl.js b/node_modules/@mtcute/core/tl/inner-tl.js
index d03b7cb..5a9eb1c 100644
--- a/node_modules/@mtcute/core/tl/inner-tl.js
+++ b/node_modules/@mtcute/core/tl/inner-tl.js
@@ -74,7 +74,8 @@ RpcError.fromTl = function(obj) {
   } else if ((match = err.text.match(/^USER_MIGRATE_(\d+)$/)) != null) {
     err.text = "USER_MIGRATE_%d";
     err.newDc = parseInt(match[1]);
-  } else return err;
+  }
+  return err;
 };
 RpcError.BAD_REQUEST = 400;
 RpcError.UNAUTHORIZED = 401;
